java/src/main/java/com/saucelabs/simplesauce/SauceOptions.java

```diff
@@ -1,134 +1,149 @@
+    @Getter @Setter private String unhandledPromptBehavior;
+
+    // Sauce Settings
+    @Getter @Setter private Boolean avoidProxy = null;
+    @Setter private String build;
+    @Getter @Setter private Boolean capturePerformance = null;
+    @Getter @Setter private String chromedriverVersion;
+    @Getter @Setter private Integer commandTimeout = null;
+    @Getter @Setter private Map customData = null;
+    @Getter @Setter private Boolean extendedDebugging = null;
+    @Getter @Setter private Integer idleTimeout = null;
+    @Getter @Setter private String iedriverVersion;
+    @Getter @Setter private Integer maxDuration = null;
+    @Getter @Setter private String name;
+    @Getter @Setter private String parentTunnel;
+    @Getter @Setter private Map prerun;
+    @Getter @Setter private String prerunUrl;
+    @Getter @Setter private Integer priority = null;
+    @Getter @Setter private String publicRestricted;
+    @Getter @Setter private Boolean recordLogs = null;
+    @Getter @Setter private Boolean recordScreenshots = null;
+    @Getter @Setter private Boolean recordVideo = null;
+    @Getter @Setter private String screenResolution;
+    @Getter @Setter private String seleniumVersion;
+    @Getter @Setter private List tags = null;
+    @Getter @Setter private String timeZone;
+    @Getter @Setter private String tunnelIdentifier;
+    @Getter @Setter private Boolean videoUploadOnPass = null;
+
+    public static <E extends Enum<E>> boolean isInEnum(String value, Class<E> enumClass) {
+        return Arrays.stream(enumClass.getEnumConstants()).anyMatch(e -> e.name().equals(value));
+    }
+
+    public SauceOptions() {
+        this(new MutableCapabilities());
+    }
+
+    public SauceOptions(MutableCapabilities capabilities) {
+        seleniumCapabilities = capabilities;
+        if (capabilities.getCapability("browserName") != null) {
+            browserName = (String) capabilities.getCapability("browserName");
+        }
+    }
+
+    public MutableCapabilities toCapabilities() {
+        MutableCapabilities w3cCapabilities = getSeleniumCapabilities();
+        MutableCapabilities sauceCapabilities = new MutableCapabilities();
+
+        Arrays.stream(this.getClass().getDeclaredFields()).forEach(field -> {
+            try {
+                String fieldName = field.getName();
+                if (!fieldName.equals("seleniumCapabilities")) {
+                    Object value = getCapability(fieldName);
+                    String key = fieldName.equals("prerunUrl") ? "prerun" : fieldName;
+                    if (value == null) {
+                        // noop
+                    } else if (isInEnum(key, Options.W3C.class)) {
+                        w3cCapabilities.setCapability(fieldName, value);
+                    } else if (isInEnum(key, Options.SAUCE.class)) {
+                        sauceCapabilities.setCapability(fieldName, value);
+                    } else if (key.equals("publicRestricted")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4721029)